packages/cli/src/executions/execution-recovery.service.ts

```diff
@@ -42,12 +58,89 @@ export class ExecutionRecoveryService {
 		return amendedExecution;
 	}
 
+	/**
+	 * Schedule a cycle to mark dangling executions as crashed in queue mode.
+	 */
+	scheduleQueueRecovery(waitMs = this.queueRecoverySettings.waitMs) {
```

Comment:
Friday brain questions: Is there a chance this could cause a memory leak? As we're doing almost like a recursion, is it possible that this could cause the closure to accumulate? 🤔

Reply:
Discussion here: https://n8nio.slack.com/archives/C069HS026UF/p1718380096971489

Reply:
This could create a leak, but since there is no state in this closure, I don't think there is anything to worry.
Since `wait` isn't used inside the handler, and `this` doesn't need to be kept in the closure, I'd say this code is okay.